### ISSUE TITLE
Proxy settings

### ISSUE THREAD

**Issue by KingB0X** [ImY0mu/SimpleMMO-Hub-Dist#21]
would be nice to have a dedicated proxy settings

**Comment by ImY0mu** [comment]
Could you please detail real life use that would require this within the App Hub?

**Comment by KingB0X** [comment]
In my personal case, my work use a http proxy server and would be amazing that I could configure it directly in the hub, the windows settings are blocked by administrative rights... Sorry about my bad English

**Comment by ImY0mu** [comment]
I have not used proxy before so I am not fully knowledgeable about what it requires. 

Whenever you use it:
- Do you just enter the IP address which should be used?
- Does it also require to be able to select which protocol to use?
- Do you have to authenticate in any way?

I am currently looking into it in my free time, but it is going to take time and I am not even sure if I will get it work correctly.

**Comment by KingB0X** [comment]
-yes ip and port
-in my case it's just http but for some people maybe be socks 
-yes username and password
**thank you for your time, you're amazing**

**Comment by ImY0mu** [comment]
If I were to give you a test version, do you have some way to verify if it is working at your end?

**Comment by ImY0mu** [comment]
Test version, uploaded it to dropbox. ([Download here](https://www.dropbox.com/s/42cjtcvtnlwf9k8/SimpleMMO%20Hub%20v2%20Setup%200.0.51-22.11.8.exe?dl=0))

When I tried it with some random free proxy, I managed to get into the game, but it was not able to do any API calls, so I hope it works for you.

You can find the proxy settings at Hub's settings, gotta press the Apply button each time you launch App hub or do any change.

Lemme know what it does for you.

Thank you.

**Comment by KingB0X** [comment]
Thanks, you're awesome

**Comment by ImY0mu** [comment]
Lemme please know if it works 👀

**Comment by KingB0X** [comment]
I can't access the options, it would be better if those options were accessible before logging in to simple mmo.

**Comment by ImY0mu** [comment]
Have you downloaded it from the link above? It might want to auto update to the recent update, but it should be available as long as you do not restart it,.

**Comment by ImY0mu** [comment]
> I can't access the options, it would be better if those options were accessible before logging in to simple mmo.

Are you saying that you cannot get to the settings because you are unable to login?

**Comment by ImY0mu** [comment]
You could try to open the console with Ctrl + Shift + I, edit the below and enter it:
`window.background.changeProxy('http', '127.0.0.1', '8080', { username: 'Unknown', password: '' });`

**Comment by ImY0mu** [comment]
Closed as the issue has not received any reply in a while.
